sdfg_utils.js

```diff
@@ -196,19 +196,27 @@ function traverse_sdfg_scopes(sdfg, func, post_subscope_func=null) {
     scopes_recursive(sdfg, sdfg.nodes());
 }
 
+// contains already visited edges to speed up computation of memlet trees
+var memlet_tree_edges_visited = [];
```

Comment:
How does that work when a new graph is loaded? Might be better to pass in a state here, or store the information directly in `graph`.

Reply:
It works because a boolean is passed, indicating whether the graph has changed or not so that this array is reset. Anyway, it is not a good solution, so now this is passed as an argument, and all the memlet trees are stored in SDFGRenderer and are recomputed if the graph changes (new graph loaded, states collapsed, etc.)